src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -11893,43 +11893,308 @@ static bool Dummy(object y, object z)
         }
 
         [Fact]
-
-        public void NullableFromGenericOperatorIs()
+        public void PatternMatchingNumericHelpers()
         {
             var source = @"
-using System;
-class Program
+public static class Program
 {
-    static void Main(string[] args)
+    public static void Main()
     {
-        Console.WriteLine(((byte)1) is Odd<byte>());
-        Console.WriteLine(((byte)1) is Odd<byte?>());
-        Console.WriteLine(((byte)2) is Odd<byte>());
-        Console.WriteLine(((byte)2) is Odd<byte?>());
-        Console.WriteLine(string.Empty is Odd<string>());
-    }
-}
-public class Odd<T>
-{
-    public static bool operator is(T o)
-    {
-        switch ((object)o)
-        {
-            case byte i:    return (i % 2) != 0;
-            case sbyte i:   return (i % 2) != 0;
-            default:        return false;
-        }
+        AssertTrue(Box((byte)(0)) is (byte)(0));
+        AssertTrue(Box((byte)(0)) is (sbyte)(0));
+        AssertTrue(Box((byte)(0)) is (short)(0));
```

Comment:
I'd like to see a smaller testcase with couple common patterns with IL compare. - just to see the code quality and that it does not regress in the future.